client/src/lib.rs

```diff
@@ -153,50 +256,118 @@ impl TurnkeyClient {
         }
     }
 
-    pub fn current_timestamp(&self) -> String {
+    /// Returns the current timestamp as a u128.
+    pub fn current_timestamp(&self) -> u128 {
         let now = SystemTime::now();
 
         now.duration_since(UNIX_EPOCH)
             .expect("Time went backwards")
             .as_millis()
-            .to_string()
+    }
+
+    /// Processes a `Request` (at `path`) by:
+    /// * Serializing the request to JSON
+    /// * Signing the request with the client's API key
+    /// * POSTing the POST body and the associated stamp to the Turnkey API
+    ///
+    /// This function is generic and can handle POSTing queries or activities.
+    pub async fn process_request<Request, Response>(
+        &self,
+        request: &Request,
+        path: String,
+    ) -> Result<Response, TurnkeyClientError>
+    where
+        Request: Serialize,
+        Response: DeserializeOwned,
+    {
+        let url = format!("{}{}", self.base_url, path);
+        let post_body = serde_json::to_string(&request)?;
+        let stamp = self.api_key.stamp(post_body.clone())?;
+        let res = self
+            .http
+            .post(url)
+            .header("X-Stamp", stamp)
+            .body(post_body)
+            .send()
+            .await?;
+
+        let status = res.status();
+        let content_type = res
+            .headers()
+            .get(CONTENT_TYPE)
+            .ok_or(TurnkeyClientError::MissingContentTypeHeader)?
+            .to_str()
+            .map_err(|e| TurnkeyClientError::HeaderToStrError(e.to_string()))?
+            .parse::<mime::Mime>()
+            .map_err(|e| TurnkeyClientError::HeaderFromStrError(e.to_string()))?;
+        let text = res.text().await?;
+
+        if !status.is_success() {
+            return Err(TurnkeyClientError::UnexpectedHttpStatus(
+                status.as_u16(),
+                text,
+            ));
+        }
+
+        if content_type != mime::APPLICATION_JSON {
+            return Err(TurnkeyClientError::UnexpectedMimeType(
+                content_type.to_string(),
+            ));
```

Comment:
You may want to log e.g. the first 200 bytes of the response as debug level logging?
It can be useful for weirder connectivity problems e.g. the user is behind a HTTP(S) proxy; or on captive portal wifi  that starts returning "you must be logged in" or something.

Reply:
Good idea! I've logged this in https://github.com/tkhq/rust-sdk/issues/14 -- will spin up a separate branch/PR for it.